### PR TITLE
fix: wrong pages number which includes private repository count.

### LINKED ISSUE #843: Wrong pages number which include private repository count
## Description

Without logging. if someone access to the personal profile page directly (eg. https://try.gitea.io/appleboy). Wrong pages number which includes private repository count.

screen shot 2017-02-05 at 1 40 46 pm

@lunny  Please assign this issue to me.
...

### PR DESCRIPTION
**Pull request by appleboy**
fix #843

* Add unit testing.
* Add `GetPublicRepositoryCount` func
* Add `GetPrivateRepositoryCount` func

### CODE REVIEW

**Comment by lunny** [comment]
trusted LGTM

File: models/repo.go [hunk 1]
```diff
@@ -1734,11 +1734,29 @@ func getRepositoryCount(e Engine, u *User) (int64, error) {
...
 func GetRepositoryCount(u *User) (int64, error) {
 	return getRepositoryCount(x, u)
 }
 
+// GetPublicRepositoryCount returns the total number of public repositories of user.
+func GetPublicRepositoryCount(u *User) (int64, error) {
+	return getPublicRepositoryCount(x, u)
+}
+
+// GetPrivateRepositoryCount returns the total number of public repositories of user.
```
Comment (ethantkoenig) [inline models/repo.go]: public -> private

File: models/repo_test.go [hunk 2]
```diff
@@ -68,3 +73,27 @@ func TestRepo(t *testing.T) {
 		})
 	})
 }
+
+func TestGetRepositoryCount(t *testing.T) {
+	assert.NoError(t, PrepareTestDatabase())
+
+	count, _ := GetRepositoryCount(&User{ID: int64(10)})
```
Comment (ethantkoenig) [inline models/repo_test.go]: Could we assert that the returned `error` is `nil`, instead of just ignoring it?

File: models/repo_test.go [hunk 3]
```diff
@@ -68,3 +73,27 @@ func TestRepo(t *testing.T) {
...
+	privateCount, _ := GetPrivateRepositoryCount(&User{ID: int64(10)})
+	publicCount, _ := GetPublicRepositoryCount(&User{ID: int64(10)})
+	assert.Equal(t, int64(3), count)
+	assert.Equal(t, (privateCount + publicCount), count)
+}
+
+func TestGetPublicRepositoryCount(t *testing.T) {
+	assert.NoError(t, PrepareTestDatabase())
+
+	count, _ := GetPublicRepositoryCount(&User{ID: int64(10)})
```
Comment (ethantkoenig) [inline models/repo_test.go]: ^

File: models/repo_test.go [hunk 4]
```diff
@@ -68,3 +73,27 @@ func TestRepo(t *testing.T) {
...
+	assert.Equal(t, int64(1), count)
+}
+
+func TestGetPrivateRepositoryCount(t *testing.T) {
+	assert.NoError(t, PrepareTestDatabase())
+
+	count, _ := GetPrivateRepositoryCount(&User{ID: int64(10)})
```
Comment (ethantkoenig) [inline models/repo_test.go]: ^

File: routers/user/profile.go [hunk 5]
```diff
@@ -134,14 +134,23 @@ func Profile(ctx *context.Context) {
...
 				return
 			}
 			ctx.Data["Repos"] = repos
-			ctx.Data["Page"] = paginater.New(ctxUser.NumRepos, setting.UI.User.RepoPagingNum, page, 5)
-			ctx.Data["Total"] = ctxUser.NumRepos
+
+			if showPrivate {
+				total = ctxUser.NumRepos
+			} else {
+				count, _ := models.GetPublicRepositoryCount(ctxUser)
```
Comment (ethantkoenig) [inline routers/user/profile.go]: Could we check if the returned `error` is `nil` instead of ignoring it?

**Review by ethantkoenig — CHANGES_REQUESTED** [review]

File: models/repo.go (hunk 1 above)
Comment (appleboy) [reply models/repo.go]: Why? Already has `getPrivateRepositoryCount`  private function.

Comment (ethantkoenig) [reply models/repo.go]: `returns the total number of public repositories of user.` -> `returns the total number of private repositories of user.` :smile:

Comment (appleboy) [reply models/repo.go]: @ethantkoenig understood. 👍

Comment (appleboy) [reply models/repo.go]: Done

File: models/repo_test.go (hunk 2 above)
Comment (appleboy) [reply models/repo_test.go]: Done

File: models/repo_test.go (hunk 3 above)
Comment (appleboy) [reply models/repo_test.go]: Done

File: models/repo_test.go (hunk 4 above)
Comment (appleboy) [reply models/repo_test.go]: Done

File: routers/user/profile.go (hunk 5 above)
Comment (appleboy) [reply routers/user/profile.go]: Done

**Review by ethantkoenig — APPROVED** [review]
LGTM

**Comment by ethantkoenig** [comment]
LGTM

**Pull request merged**